src/EVM/SymExec.hs

```diff
@@ -762,47 +807,50 @@ verifyResults preState res = (map snd res, fmap toVRes res)
       Error e -> Error e
       Qed -> Qed
 
--- | Symbolically execute the VM and find possible inputs given the
--- postcondition, if available.
-verifyInputs
+-- | Symbolically execute the VM with optional custom handler for immediate Cex processing
+verifyInputsWithHandler
   :: App m
   => SolverGroup
   -> VeriOpts
   -> Fetch.Fetcher Symbolic m
   -> VM Symbolic
   -> Postcondition
+  -> Maybe (VM Symbolic -> SMTResult -> Expr End -> m ())
   -> m ([(SMTResult, Expr End)], [(PartialExec, Expr End)])
-verifyInputs solvers opts fetcher preState post = do
+verifyInputsWithHandler solvers opts fetcher preState post cexHandler = do
   conf <- readConfig
   let call = mconcat ["prefix 0x", getCallPrefix preState.state.calldata]
   when conf.debug $ liftIO $ putStrLn $ "   Exploring call " <> call
 
-  ends <- executeVM fetcher opts.iterConf preState
-
-  let partials = getPartials ends
+  results <- executeVM fetcher opts.iterConf preState $ \leaf -> do
+    -- Extract partial if applicable
+    let mPartial = case leaf of
+          Partial _ _ p -> Just (p, leaf)
+          _ -> Nothing
+
+    -- Check if this leaf needs SMT checking
+    let props = toProps leaf preState post
+    smtResult <- if canBeSat (props, leaf)
+      then do
+        res <- checkSatWithProps solvers props
+        when (conf.debug && conf.verb >=2) $ liftIO $ putStrLn $ "   Checking leaf with props: " <> show props <> " SMT result: " <> show res
+        -- Call custom handler if provided (for immediate Cex processing/validation/printing)
+        case (cexHandler, res) of
+          (Just handler, cex@(Cex _)) -> handler preState cex leaf
+          _ -> pure ()
+        pure (res, leaf)
+      else pure (Qed, leaf)
+    pure (smtResult, mPartial)
+
+  let (smtResults, partials) = unzip results
   when conf.debug $ liftIO $ do
-    printPartialIssues ends ("the call " <> call)
-    putStrLn $ "   Exploration finished, " <> show (length ends) <> " branch(es) to check in call " <> call <> " of which partial: "
-                <> show (length partials)
-    when (conf.verb >=2) $ putStrLn $ "   results:\n" <> prettyvmresults ends
+    putStrLn $ "   Exploration finished, " <> show (length results) <> " branch(es) checked in call " <> call <> " of which partial: "
+                <> show (length smtResults)
```

Comment:
I think this message here is misleading now.
At this point, not only exploration has finished, but also all the solving, no?
At least before, "exploration" meant only the first phase, excluding the solving.